src/cli/generator-html.ts

```diff
@@ -333,23 +333,20 @@ function generateExpression(expression: Expression | SimpleExpression, ctx: Gene
         return expression.value
     }
     else if (isSymbolReference(expression)) {
-        let value = ''
-        ctx.argumentStack[0].forEach(function (el) {
-            if ((el as any).name === expression.symbol.ref?.name) {
-                value = (el as any).value
-            }
-        })
-        return value
+        const values = ctx.argumentStack[0];
+        const lastIndex = values
+            .reduce<number>((maxIndex, el, currentIndex) => (el as any).name === expression.symbol.ref?.name ? currentIndex : maxIndex, -1);
+        return lastIndex === -1 ? '' : values[lastIndex] as string;
```

Comment:
basically you want to have only the last element here but you perform an assignment to a variable N times, where N is the length of the `argumentStack[0]`.

The reduce function is one way to transform implicit information stored in an array (here: the max index that obeys a certain condition).